atomic_reactor/util.py

```diff
@@ -1557,3 +1558,40 @@ def has_operator_manifest(workflow):
     except KeyError:
         operator_label = 'false'
     return operator_label.lower() == 'true'
+
+
+def annotation(*keys):
+    """
+    Annotate an instance method of a `BuildPlugin`. When that method executes,
+    its result will be stored in the plugin's workflow under the given keys.
+
+    Example:
+    >>> from atomic_reactor.plugin import BuildPlugin
+
+    >>> class MyBuildPlugin(BuildPlugin):
+    >>>     key = 'my_build_plugin'
+    >>>
+    >>>     @annotation('foo', 'bar')
+    >>>     def run(self, x):
+    >>>         return x
+
+    >>> p = MyBuildPlugin(...)
+    >>> p.run(1)
+    >>> p.workflow.annotations  # {'foo': [1], 'bar': [1]}
+
+    :param keys: Keys to annotate the method with
+    :return: Decorator that will turn the method into an annotated one
+    """
+
+    def annotate(build_plugin_method):
+        @functools.wraps(build_plugin_method)
+        def annotated(self, *args, **kwargs):
+            result = build_plugin_method(self, *args, **kwargs)
+            for key in keys:
+                prev_results = self.workflow.annotations.setdefault(key, [])
```

Comment:
Do we want to allow multiple plugins to append to the same annotation? Shouldn't this rather fail when updating existing annotation?

Reply:
That's what I don't know, but your guess is probably better than mine

Reply:
@twaugh what do you think?

Reply:
Let's make it fail. If we decide later it would be useful to combine multiple results we can change it then.

Reply:
ok, done